File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        jumps = 0
        farthest = 0
        end = 0
        for i in range(len(nums)-1):
            farthest = max(farthest, i+nums[i])
            if i == end:
                jumps += 1
                end = farthest
        return jumps

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int jumps = 0;
        int farthest = 0;
        int current_end = 0;
        for(int i=0;i<nums.length-1;i++){
            farthest = Math.max(farthest, i+nums[i]);
            if(i == current_end){
                jumps+=1;
                current_end = farthest;
            }
        }
        return jumps;
    }
}